okhttp/src/main/kotlin/okhttp3/internal/connection/RealCall.kt

```diff
@@ -232,16 +233,17 @@ class RealCall(
 
   /** Finds a new or pooled connection to carry a forthcoming request and response. */
   internal fun initExchange(chain: RealInterceptorChain): Exchange {
-    check(!noMoreExchanges) { "released" }
-    check(exchange == null)
+    check(expectMoreExchanges) { "released" }
+    check(!responseBodyOpen)
+    check(!requestBodyOpen)
```

Comment:
these two values being false imply that the exchange is null, but this is more semantic